sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -146,12 +120,28 @@ public static BinaryData fromStream(InputStream inputStream) {
      * @return A {@link Mono} of {@link BinaryData} representing the {@link Flux} of {@link ByteBuffer}.
      */
     public static Mono<BinaryData> fromFlux(Flux<ByteBuffer> data) {
-        if (Objects.isNull(data)) {
-            return Mono.just(EMPTY_DATA);
-        }
+        return fromFlux(data, null);
+    }
 
-        return FluxUtil.collectBytesInByteBufferStream(data)
-            .flatMap(bytes -> Mono.just(new BinaryData(bytes)));
+    /**
+     * Creates an instance of {@link BinaryData} from the given {@link Flux} of {@link ByteBuffer}.
+     * <p>
+     * If the {@code data} is null an empty {@link BinaryData} will be returned.
```

Comment:
This statement isn't true anymore, there is an `Objects.requireNonNull(data)` call.